server/aesdsocket.c

```diff
@@ -213,217 +608,410 @@ int main(int argc, char *argv[])
+    pthread_t timer_thread_id;
+    pthread_create(&timer_thread_id, NULL, timer_routine, &mutex);
+
+    while (1)
+    {
+        cli_len = sizeof(client);
+
+        cli_fd = accept(sock_fd, (struct sockaddr *)&client, &cli_len);
+
+        if (cli_fd < 0)
+        {
+            printf("Error in accept. Error code:%d", errno);
+            syslog(LOG_ERR, "Error in accept");
+        }
+        else
+        {
+            printf("Connection accepted\n");
+            syslog(LOG_ERR, "New Connection accepted");
+        }
+
+        // Printing IP address of client
+        char address[30]; // To store the IP address converted into string
+
+        // const char *inet_ntop(int af, const void *restrict src,
+        const char *ip_addr = inet_ntop(AF_INET, &client.sin_addr, address, sizeof(address));
+        int client_port = htons(client.sin_port);
+        if (ip_addr == NULL)
+        {
+            printf("Error in obtaining ip address of the client\r\n");
+        }
+        else
+        {
+            printf("Accepted connection from %s %d\r\n", ip_addr, client_port);
+            // Log this info
+            syslog(LOG_INFO, "Accepted connection from %s %d\r\n", ip_addr, client_port);
+        }
+
+        threads_node = (node *)malloc(sizeof(node));
+        if(threads_node == NULL)
+        {
+              printf("Error in obtaining ip address of the client\r\n");
+              graceful_exit();
+        }
+        SLIST_INSERT_HEAD(&head, threads_node, entry);
+        threads_node->client_parameters.client_fd = cli_fd;
+        threads_node->client_parameters.address = client;
+        threads_node->client_parameters.mutex = &mutex;
+        threads_node->client_parameters.thread_complete = false;
+        
+        pthread_create(&(threads_node->client_parameters.thread_id), NULL, thread_routine, (void*)&threads_node->client_parameters);
+
+        SLIST_FOREACH(threads_node, &head, entry)
+        {
+            if(threads_node->client_parameters.thread_complete == true)
+            {
+                pthread_join(threads_node->client_parameters.thread_id, NULL);
+                //after completion close the connection 
+                shutdown(threads_node->client_parameters.client_fd, SHUT_RDWR);
+                close(threads_node->client_parameters.client_fd);
+                printf("Closed connection from %s %d\r\n", ip_addr, client_port);
+                // Log this info
+              //  syslog(LOG_INFO, "Closed connection from %s %d\r\n", ip_addr, client_port);    
+            }
+        }
+
+    }
+    pthread_join(timer_thread_id, NULL);
+    while(!SLIST_EMPTY(&head))
+    {
+        threads_node = SLIST_FIRST(&head);
+        pthread_cancel(threads_node->client_parameters.thread_id);
+        SLIST_REMOVE_HEAD(&head, entry);
+       // if(threads_node != NULL)
+       // {
+        free(threads_node);
+       // }
+        threads_node = NULL;
+    }
+    graceful_exit();
+    remove(FILE_PATH);
+
+   /*
```

Comment:
@adna4700 delete code rather than commenting out.  Use revision control to track changes or revert as needed.